pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedPrivateFieldRule.java

```diff
@@ -121,4 +136,28 @@ private boolean actuallyUsed(List<NameOccurrence> usages) {
     private boolean isOK(String image) {
         return "serialVersionUID".equals(image) || "serialPersistentFields".equals(image) || "IDENT".equals(image);
     }
+
+    /**
+     * Checks whether the given node is annotated with annotation in the set.
+     * The node should be field declaration.
+     *
+     * @param node
+     *            the node to check
+     * @return <code>true</code> if the annotation has been found
+     */
+    protected boolean hasNeglectAnnotation(Node node) {
+        boolean result = false;
+        Node parent = node.jjtGetParent();
+        List<ASTAnnotation> annotations = parent.findChildrenOfType(ASTAnnotation.class);
+        for (ASTAnnotation annotation : annotations) {
+            ASTName name = annotation.getFirstDescendantOfType(ASTName.class);
+            if (name != null) {
+                String annotationName = name.getImage();
+                if (NEGLECT_ANNOTATIONS.contains(annotationName)) {
```

Comment:
This would fail with `@javafx.fxml.FXML` for example. The typical way to consider both cases is to compare the type of the annotation instead of its name, i.e., assuming your set is now a `Set<Class<?>>` containing these types:

```java
NEGLECTED_ANNOTATION_TYPES.contains(((TypeNode) annotation.jjtGetChild(0)).getType())
```

To note that PR #912 will add new methods to smooth the interface of Annotation among other nodes, which would allow to simplify the ugly child access and cast to `annotation.getType()`.

Reply:
Good idea!
Done.